torch/_inductor/lowering.py

```diff
@@ -1999,21 +1999,30 @@ def inner_fn(index):
 
 @register_lowering(aten.bucketize, type_promotion_kind=None)
 def bucketize(
-    input: TensorBox,
+    input: Union[TensorBox, float, int],
     boundaries: TensorBox,
     *,
     out_int32: bool = False,
     right: bool = False,
 ):
     assert len(boundaries.get_size()) == 1
 
+    # First check if input is not a tensor in which case fallback to
+    # aten.bucketize.Scalar
+    if isinstance(input, Number):
+        return fallback_handler(aten.bucketize.Scalar, add_to_fallback_set=False)(
+            input, boundaries, out_int32=out_int32, right=right
+            )
+
     if not (
         V.graph.has_feature(input, BackendFeature.BUCKETIZE)
         and V.graph.has_feature(boundaries, BackendFeature.BUCKETIZE)
     ):
-        return fallback_handler(aten.bucketize.Tensor, add_to_fallback_set=False)(
+        if isinstance(input, TensorBox):
```

Comment:
Not sure I understand the logic change here?

Reply:
Changed the logic in the if statement back since it should be left as it was before. Will add in a test later.